### ISSUE TITLE
Incorrect behavior of for on MutMap

### ISSUE THREAD

**Issue by meirdev** [winglang/wing#5625]
### I tried this:

```wing
class A {
  foo: MutMap<str>;

  new() {
    this.foo = MutMap<str>{};
  }

  pub inflight add(s: str) {
    this.foo.set(s, s);
  }

  pub inflight loop() {
    log("{Json.stringify(this.foo)}");

    for i in this.foo.keys() {
      log("HERE {i}");
    }
  }
}

let a = new A();

test "" {
  a.add("test1");
  a.add("test2");

  a.loop();
}
```

### This happened:

the output is:

```
{"test1":"test1","test2":"test2"}
```

### I expected this:

the output to be:

```
{"test1":"test1","test2":"test2"}
HERE test1
HERE test2
```

### Anything else?

The `MutMap.keys` macro doesn't seem to work properly (or something else happens later) :

```javascript
module.exports = function({ $std_Json }) {
  class A {
    constructor({ $Object_keys_this_foo_, $this_foo }) {
      this.$Object_keys_this_foo_ = $Object_keys_this_foo_;
      this.$this_foo = $this_foo;
    }
    async add(s) {
      ((obj, args) => { obj[args[0]] = args[1]; })(this.$this_foo, [s, s]);
    }
    async loop() {
      console.log(String.raw({ raw: ["", ""] }, ((json, opts) => { return JSON.stringify(json, null, opts?.indent) })(this.$this_foo)));
      for (const i of this.$Object_keys_this_foo_) {
        console.log(String.raw({ raw: ["HERE ", ""] }, i));
      }
    }
  }
  return A;
}
```

Note the variable: `this.$Object_keys_this_foo_`

### Wing Version

0.57.4

### Node.js Version

v20.11.0

### Platform(s)

MacOS

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by meirdev** [comment]
Same result even when I don't use a for loop:

```wing
class A {
  foo: MutMap<str>;

  new() {
    this.foo = MutMap<str>{};
  }

  pub inflight add(s: str) {
    this.foo.set(s, s);
  }

  pub inflight loop() {
    log("{Json.stringify(this.foo)} {this.foo.keys()}");
  }
}
```

**Comment by tsuf239** [comment]
It seems to be a problem with the lifting- when you make `foo` an inflight property, the keys are logged correctly.
```
class A {
  inflight foo: MutMap<str>;

  inflight new() {
    this.foo = MutMap<str>{};
  }

  pub inflight add(s: str) {
    this.foo.set(s, s);
  }

  pub inflight loop() {
    log("{Json.stringify(this.foo)}");

    for i in this.foo.keys() {
      log("HERE {i}");
    }
  }
}
```

**Comment by yoav-steinberg** [comment]
The problem is that the code mutates a lifted map. Allowing this was a bug fixed in #6303.
Now the original code won't compile. You'll need to either make `foo` an inflight field (like @tsuf239 did above 👆) or create a mutable copy of the lifted `foo` inflight (`foo.copyMut()`).

I'm closing this. Let me know if something isn't clear or can be improved.
